Realm/Realm.Weaver/RealmWeaver.cs

```diff
@@ -234,6 +253,118 @@ Analytics payload
             return WeaveModuleResult.Success(weaveResults);
         }
 
+        private WeaveTypeResult WeaveGeneratedType(TypeDefinition type)
+        {
+            _logger.Debug("Weaving generated " + type.Name);
+
+            var interfaceName = $"I{type.Name}Accessor";
+            var interfaceType = _moduleDefinition.GetType("Realms.Generated", interfaceName);
+
+            var persistedProperties = new List<WeavePropertyResult>();
+
+            // We need to weave all (and only) the properties in the accessor interface
+            foreach (var interfaceProperty in interfaceType.Properties)
+            {
+                var prop = type.Properties.First(p => p.Name == interfaceProperty.Name);
+
+                try
+                {
+                    var weaveResult = WeaveGeneratedClassProperty(prop, interfaceType);
+                    persistedProperties.Add(weaveResult);
+                }
+                catch (Exception e)
+                {
+                    var sequencePoint = prop.GetSequencePoint();
+                    _logger.Error(
+                        $"Unexpected error caught weaving property '{type.Name}.{prop.Name}': {e.Message}.\r\nCallstack:\r\n{e.StackTrace}",
+                        sequencePoint);
+
+                    return WeaveTypeResult.Error(type.Name, isGenerated: true);
+                }
+            }
+
+            return WeaveTypeResult.Success(type.Name, persistedProperties, isGenerated: true);
+        }
+
+        private WeavePropertyResult WeaveGeneratedClassProperty(PropertyDefinition prop, TypeDefinition interfaceType)
```

Comment:
I believe this can also remove the compiler generated backing fields?

Reply:
Most probably yes, I remember we talked about it. I'll add another ticket for this so I don't forget